### PR TITLE
sync with coco

### PR DESCRIPTION
**Pull request by holdenkold**

### CODE REVIEW

File: cma_fmin.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+from cma import CMA
+from vector_initializer import XavierInitializer
+
+def fmin(problem, mean, sigma):
+    initializer = XavierInitializer(30)
+    # Initialize CMA-ES with the provided parameters
+    es = CMA(mean=mean, sigma=sigma)
```
Comment (Michalski-Artur) [inline cma_fmin.py]: Don't we also need to inject initializer here? Or you are running tests with zero initializer?

File: example_experiment2.py
Comment (Michalski-Artur) [inline example_experiment2.py:1]: I have just a couple of questions to verify my understanding:
1. I guess this is the code responsible for actually running tests and generating report? 
2. And we inject evaluator (cmaes algorithm) through fmin in cma_fmin file? 
3. I am wondering where are we actually providing list or single function to run tests on? I can see they are aggregated in `suite`, but how they are fetched? Or maybe coco has predefined set of tests here and it fully covers our needs?

Also I would appreciate if you could add some description and sample call for running tests for selected functions in PR description or readme

File: cma_fmin.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        # Generate new candidate solutions using the ask() method
+        solutions = [es.ask() for _ in range(es.population_size)]
+
+        # Evaluate the objective function for each candidate solution
+        solutions_eval = [(sol, problem(sol)) for sol in solutions]
+        # Provide the evaluation results to the tell() method
+        es.tell(solutions_eval)
+
+    # Retrieve the best solution and the ESResult object
+    xopt = es._best[0]
```
Comment (Michalski-Artur) [inline cma_fmin.py:15]: So for some reason, coco also needs to know about global best solution found?

**Review by Michalski-Artur — APPROVED** [review]
Looks good 👍 

Just from curiosity and to plan future work: How long does it roughly take for tests to execute?

File: cma_fmin.py (hunk 1 above)
Comment (holdenkold) [reply cma_fmin.py]: for now we using default(zero) initializer. I will delete declaration of XavierInitializer as we are not using it.

File: example_experiment2.py
Comment (holdenkold) [reply example_experiment2.py:1]: 1. yes, this is the code generated from coco library. I just copied it to our repository to not create coco repository dependancy. After tests are done report is generated automatically
2. we inject evaluator by switching `fmin` assignment in this file in line 90 and 168.
3. I'm not sure I fully understand a question. But coco has predefined set of fitness functions in benchmark bbob which is used in this experiment. Example of the functions used can be find on the image 
image
